x-pack/plugin/logsdb/src/test/java/org/elasticsearch/xpack/logsdb/LegacyLicenceIntegrationTests.java

```diff
@@ -0,0 +1,146 @@
+import org.elasticsearch.index.mapper.SourceFieldMapper;
+import org.elasticsearch.license.AbstractLicensesIntegrationTestCase;
+import org.elasticsearch.license.GetFeatureUsageRequest;
+import org.elasticsearch.license.GetFeatureUsageResponse;
+import org.elasticsearch.license.License;
+import org.elasticsearch.license.LicenseService;
+import org.elasticsearch.license.LicensedFeature;
+import org.elasticsearch.license.TransportGetFeatureUsageAction;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xpack.core.LocalStateCompositeXPackPlugin;
+import org.hamcrest.Matcher;
+import org.junit.Before;
+
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.Collection;
+import java.util.List;
+
+import static org.elasticsearch.test.ESIntegTestCase.Scope.TEST;
+import static org.elasticsearch.xpack.logsdb.SyntheticSourceLicenseServiceTests.createEnterpriseLicense;
+import static org.elasticsearch.xpack.logsdb.SyntheticSourceLicenseServiceTests.createGoldOrPlatinumLicense;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+
+@ESIntegTestCase.ClusterScope(scope = TEST, numDataNodes = 1, numClientNodes = 0, supportsDedicatedMasters = false)
+public class LegacyLicenceIntegrationTests extends AbstractLicensesIntegrationTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(P.class);
+    }
+
+    @Before
+    public void setup() throws Exception {
+        wipeAllLicenses();
+        ensureGreen();
+        License license = createGoldOrPlatinumLicense();
+        putLicense(license);
+        ensureGreen();
+    }
+
+    public void testSyntheticSourceUsageDisallowed() {
+        createIndexWithSyntheticSourceAndAssertExpectedType("test", "STORED");
+
+        assertFeatureUsage(SyntheticSourceLicenseService.SYNTHETIC_SOURCE_FEATURE_LEGACY, nullValue());
+        assertFeatureUsage(SyntheticSourceLicenseService.SYNTHETIC_SOURCE_FEATURE, nullValue());
+    }
+
+    public void testSyntheticSourceUsageWithLegacyLicense() {
+        createIndexWithSyntheticSourceAndAssertExpectedType(".profiling-stacktraces", "synthetic");
+
+        assertFeatureUsage(SyntheticSourceLicenseService.SYNTHETIC_SOURCE_FEATURE_LEGACY, not(nullValue()));
+        assertFeatureUsage(SyntheticSourceLicenseService.SYNTHETIC_SOURCE_FEATURE, nullValue());
+    }
+
+    public void testSyntheticSourceUsageWithLegacyLicensePastCutoff() throws Exception {
+        long startPastCutoff = LocalDateTime.of(2025, 11, 12, 0, 0).toInstant(ZoneOffset.UTC).toEpochMilli();
+        putLicense(createGoldOrPlatinumLicense(startPastCutoff));
+        ensureGreen();
+
+        createIndexWithSyntheticSourceAndAssertExpectedType(".profiling-stacktraces", "STORED");
```

Comment:
We should test with synthetic?

Reply:
yes, this method creates an index with source mode set to synthetic. The last argument here is what the expected source mode should be.